include/boost/geometry/algorithms/detail/buffer/buffer_policies.hpp

```diff
@@ -127,23 +127,23 @@ struct buffer_turn_info
     robust_point_type rob_pi, rob_pj, rob_qi, rob_qj;
 #endif
 
-    int count_within;
+    signed_index_type count_within;
```

Comment:
Sure. I didn't know exactly which variables are used to count in one direction and which counts in both so I choose a safe approach.